source/common/http/filter/squash_filter.cc

```diff
@@ -0,0 +1,342 @@
+#include "common/http/filter/squash_filter.h"
+
+#include <regex>
+#include <string>
+
+#include "envoy/http/codes.h"
+
+#include "common/common/empty_string.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/logger.h"
+#include "common/common/utility.h"
+#include "common/http/message_impl.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Http {
+
+namespace {
+// The functions below are needed due to the "C++ static initialization order fiasco."
+
+static const LowerCaseString& squashHeaderKey() {
+  CONSTRUCT_ON_FIRST_USE(LowerCaseString, "x-squash-debug");
+}
+static const std::string& postAttachmentPath() {
```

Comment:
wouldn't this make envoy startup time a tiny bit slower?

Reply:
I am changing these to constants as requested. also added the regex as a constant. will change back if we decide that CONSTRUCT_ON_FIRST_USE is more appropriate to avoid the (probably tiny) startup penalty.